cmd/format-disk-cache.go

```diff
@@ -0,0 +1,287 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+
+	errors2 "github.com/minio/minio/pkg/errors"
+	"github.com/minio/minio/pkg/lock"
+)
+
+const (
+	// Represents Cache format json holding details on all other cache drives in use.
+	formatCache = "cache"
+
+	// formatCacheV1.Cache.Version
+	formatCacheVersionV1 = "1"
+
+	formatMetaVersion1 = "1"
+)
+
+// Represents the current cache structure with list of
+// disks comprising the disk cache
+// formatCacheV1 - structure holds format config version '1'.
+type formatCacheV1 struct {
+	formatMetaV1
+	Cache struct {
+		Version string `json:"version"` // Version of 'cache' format.
+		This    string `json:"this"`    // This field carries assigned disk uuid.
+		// Disks field carries the input disk order generated the first
+		// time when fresh disks were supplied.
+		Disks []string `json:"disks"`
+	} `json:"cache"` // Cache field holds cache format.
+}
```

Comment:
@poornas as discussed you can remove references to any structs that is not related for format (i.e cacheFSObjects)

Reply:
done